### PR TITLE
CORE-17818: Remove uniqueness checker duplication

### PR DESCRIPTION
**Pull request by emilybowe**
This PR includes removal of uniqueness checker duplication and refactoring necessitated by the RPC subscription introduction.

This will include an api change to remove unused uniqueness checker topic: https://github.com/corda/corda-api/pull/1371

### CODE REVIEW

**Review by blsemo — DISMISSED** [review]
LGTM

File: components/uniqueness/uniqueness-checker-impl/src/main/kotlin/net/corda/uniqueness/checker/impl/BatchedUniquenessCheckerLifecycleImpl.kt [hunk 1]
```diff
@@ -80,65 +67,37 @@ class BatchedUniquenessCheckerLifecycleImpl @Activate constructor(
...
                 dependentComponents.registerAndStartAll(coordinator)
             }
-            is StopEvent -> {
-                dependentComponents.stopAll()
-            }
             is RegistrationStatusChangeEvent -> {
-                log.info("Uniqueness checker is ${event.status}")
-
                 if (event.status == LifecycleStatus.UP) {
+                    log.trace("The status of event: $event changed to ${event.status}, starting subscription.")
```
Comment (mbrkic-r3) [inline components/uniqueness/uniqueness-checker-impl/src/main/kotlin/net/corda/uniqueness/checker/impl/BatchedUniquenessCheckerLifecycleImpl.kt]: Do we get any additional info with this log (and logs added below)? Wouldn't log at the start of function (line 70) log everything?

Comment (emilybowe) [reply components/uniqueness/uniqueness-checker-impl/src/main/kotlin/net/corda/uniqueness/checker/impl/BatchedUniquenessCheckerLifecycleImpl.kt]: Yes I've checked it - I've edited that down so it just logs that we're starting & stopping - what do you think of it now?

Comment (mbrkic-r3) [reply components/uniqueness/uniqueness-checker-impl/src/main/kotlin/net/corda/uniqueness/checker/impl/BatchedUniquenessCheckerLifecycleImpl.kt]: Assuming that `closeManagedResources` stops and closes subscription, logging "Stopping subscription." might go before that.

**Review by mbrkic-r3 — DISMISSED** [review]
LGTM

**Review by blsemo — APPROVED** [review]
Re-approved

**Pull request merged**